packages/react/menu/src/Menu.tsx

```diff
@@ -165,21 +164,24 @@ const MenuSub: React.FC<MenuSubProps> = (props: ScopedProps<MenuSubProps>) => {
     <PopperPrimitive.Root {...popperScope}>
       <MenuProvider
         scope={__scopeMenu}
-        isSubmenu={true}
         isUsingKeyboardRef={parentMenuContext.isUsingKeyboardRef}
         dir={parentMenuContext.dir}
         open={open}
         onOpenChange={handleOpenChange}
         content={content}
         onContentChange={setContent}
         onRootClose={parentMenuContext.onRootClose}
-        contentId={useId()}
-        trigger={trigger}
-        onTriggerChange={setTrigger}
-        triggerId={useId()}
         modal={false}
```

Comment:
This isn't used by the submenu right? Do you think it's worth splitting out a general menu context from a root context / sub context?

Not sure if there are other things like this one which only are used by the root?

Reply:
That’s a good point, in fact I don’t think we even need to recursively drill some of this stuff down anymore, we can just get it from root directly now.

  How about this? https://github.com/radix-ui/primitives/pull/1406/commits/13749a630b565c468cd604c6ddbb6120e60f5d72

Reply:
I'm not sure about this. I feel like it should be the other way around, ie:
- `MenuContext` contains the things that are needed/used by the entire tree
- `MenuRootContext` contains the things that are only needed by the root
- `MenuSubContext` contains the things that are only needed by a sub

So for example `modal` would be only on `MenuRootContext`, but `dir` would be on `MenuContext`.

Does that make sense to you?

Reply:
Works for me, is this what you meant? https://github.com/radix-ui/primitives/pull/1406/commits/006b3109548227b1a09a38613e4a7f02355a8106